job_bundles/diffusion/image_generation/template.yaml

```diff
@@ -0,0 +1,174 @@
+    label: Negative Prompt
+    groupLabel: Generation Settings
+  description: Negative prompt to guide what to avoid
+  default: "blurry, low quality"
+
+- name: OutputDir
+  type: PATH
+  objectType: DIRECTORY
+  dataFlow: OUT
+  userInterface:
+    control: CHOOSE_DIRECTORY
+    label: Output Directory
+    groupLabel: Output Settings
+
+- name: NumImages
+  type: INT
+  userInterface:
+    control: SPIN_BOX
+    label: Number of Images
+    groupLabel: Generation Settings
+  description: Total number of images to generate
+  default: 10
+  minValue: 1
+  maxValue: 1000
+
+- name: Width
+  type: INT
+  userInterface:
+    control: SPIN_BOX
+    label: Width
+    groupLabel: Generation Settings
+  default: 512
+  minValue: 256
+  maxValue: 1024
+
+- name: Height
+  type: INT
+  userInterface:
+    control: SPIN_BOX
+    label: Height
+    groupLabel: Generation Settings
+  default: 512
+  minValue: 256
+  maxValue: 1024
+
+- name: NumInferenceSteps
+  type: INT
+  userInterface:
+    control: SPIN_BOX
+    label: Inference Steps
+    groupLabel: Generation Settings
+  description: Number of denoising steps
+  default: 50
+  minValue: 1
+  maxValue: 150
+
+- name: GuidanceScale
+  type: STRING
+  userInterface:
+    control: LINE_EDIT
+    label: Guidance Scale
+    groupLabel: Generation Settings
+  description: Classifier-free guidance scale
+  default: "7.5"
+
+- name: Seed
+  type: STRING
+  userInterface:
+    control: LINE_EDIT
+    label: Seed
+    groupLabel: Generation Settings
+  description: Random seed for reproducibility (-1 for random)
+  default: "-1"
+
+- name: GenerateScript
+  description: Generation script
+  userInterface:
+    control: HIDDEN
+  type: PATH
+  objectType: FILE
+  dataFlow: IN
+  default: generate_image.py
+
+steps:
+- name: GenerateImages
+  parameterSpace:
+    taskParameterDefinitions:
+    - name: ImageIndex
+      type: INT
+      range: "1-{{Param.NumImages}}"
+  script:
+    actions:
+      onRun:
+        command: bash
+        args: ['-c', '{{Task.File.Run}}']
+    embeddedFiles:
+    - name: Run
+      type: TEXT
+      runnable: true
+      data: |
+        #!/bin/bash
+        set -euo pipefail
+        
+        pip install -q diffusers[torch] transformers accelerate peft safetensors
+        
+        python -u {{Param.GenerateScript}} \
+          --model-name "{{Param.ModelName}}" \
+          --lora-path "{{Param.LoRAPath}}" \
+          --prompt "{{Param.Prompt}}" \
+          --negative-prompt "{{Param.NegativePrompt}}" \
+          --width {{Param.Width}} \
+          --height {{Param.Height}} \
+          --num-inference-steps {{Param.NumInferenceSteps}} \
+          --guidance-scale {{Param.GuidanceScale}} \
+          --seed {{Param.Seed}} \
+          --image-index {{Task.Param.ImageIndex}} \
+          --output-dir "{{Param.OutputDir}}" \
+          --lora-rank {{Param.LoRARank}}
+  hostRequirements:
+    attributes:
+    - name: attr.worker.os.family
+      anyOf: [linux]
```

Comment:
Do these templates require a GPU? If so, you can add a min gpu amount of 1 like in https://github.com/aws-deadline/deadline-cloud-samples/blob/mainline/job_bundles/gsplat_pipeline/template.yaml#L249.